sklearn/neighbors/nca.py

```diff
@@ -0,0 +1,526 @@
+# coding: utf-8
+"""
+Neighborhood Component Analysis
+"""
+
+# License: BSD 3 Clause
+
+from __future__ import print_function
+
+import numpy as np
+import sys
+import time
+from scipy.misc import logsumexp
+from scipy.optimize import minimize
+from sklearn.preprocessing import OneHotEncoder
+
+from ..base import BaseEstimator, TransformerMixin
+from ..preprocessing import LabelEncoder
+from ..decomposition import PCA
+from ..utils.multiclass import check_classification_targets
+from ..utils.random import check_random_state
+from ..utils.validation import check_is_fitted, check_array, check_X_y
+from ..externals.six import integer_types
+
+
+class NeighborhoodComponentsAnalysis(BaseEstimator, TransformerMixin):
+    """Neighborhood Components Analysis
+
+    Parameters
+    ----------
+    n_features_out : int, optional (default=None)
+        Preferred dimensionality of the embedding.
+
+    init : string or numpy array, optional (default='pca')
+        Initialization of the linear transformation. Possible options are
+        'pca', 'identity', 'random', and a numpy array of shape
+        (n_features_a, n_features_b).
+
+        pca:
+            ``n_features_out`` many principal components of the inputs passed
+            to :meth:`fit` will be used to initialize the transformation.
```

Comment:
all params are not indented the same way

Reply:
This is because `pca`, `identity`, `random` and `numpy array` are not arguments but they are possible choices for argument `init`. I took the syntax from LMNN. Should I write it in another way ?